guides/ansible_engine/5-role/README.md

```diff
@@ -0,0 +1,221 @@
+
+For this exercise, you are going to take the playbook you just wrote and refactor it into a role.  In addition, you'll learn to use Ansible Galaxy.
+
+Let's begin with seeing how your apache-basic-playbook will break down into a role.
+
+![Role directory structure](roledir_1.png)
+
+Fortunately, you don't have to create all of these directories and files by hand.  That's where Ansible Galaxy comes in.
+
+## Section 1: Using Ansible Galaxy to initialize a new role
+
+Ansible Galaxy is a free site for finding, downloading, and sharing roles.  It's also pretty handy for creating them which is what we are about to do here.
+
+
+### Step 1:
+
+Navigate to your `apache-basic-playbook` project.
+
+```bash
+cd ~/apache-basic-playbook
+```
+
+
+### Step 2:
+
+Create a directory called `roles` and `cd` into it.
+
+```bash
+mkdir roles
+cd roles
+```
+
+
+### Step 3:
+
+Use the `ansible-galaxy` command to initialize a new role called `apache-simple`.
+
+```bash
+ansible-galaxy init apache-simple
+```
+
+Take a look around the structure you just created.  It should look a lot like Figure 1 above.  However, we need to complete one more step before moving onto section 2.  It is Ansible best practice to clean out role directories and files you won't be using.  For this role, we won't be using anything from `files`, `tests`.
+
+
+### Step 4:
+
+Remove the `files` and `tests` directories
+
+```bash
+cd ~/apache-basic-playbook/roles/apache-simple/
+rm -rf files tests
+```
+
+
+## Section 2: Breaking Your `site.yml` Playbook into the Newly Created `apache-simple` Role
+
+
+In this section, we will separate out the major parts of your playbook including `vars:`, `tasks:`, `template:`, and `handlers:`.
+
+### Step 1:
+
+Make a backup copy of `site.yml`, then create a new `site.yml`.
+
+```bash
+cd ~/apache-basic-playbook
+mv site.yml site.yml.bkup
+vim site.yml
+```
+
+### Step 2:
+
+Add the play definition and the invocation of a single role.
+
+```yml
+---
+- hosts: web
+  name: This is my role-based playbook
+  become: yes
+
+  roles:
+    - apache-simple
+```
+
+### Step 3:
+
+Add some default variables to your role in `roles/apache-simple/defaults/main.yml`.
+
+```yml
+---
+# defaults file for apache-simple
+apache_test_message: This is a test message
+apache_max_keep_alive_requests: 115
+```
+
+### Step 4:
+
+Add some role-specific variables to your role in `roles/apache-simple/vars/main.yml`.
+
+[source,bash]
+```
+---
+# vars file for apache-simple
+httpd_packages:
+  - httpd
+  - mod_wsgi
+```
+
+---
+**NOTE**
+####
+> Hey, wait just a minute there buster... did you just have us put variables in two seperate places?
+
+Yes... yes we did.  Variables can live in quite a few places.  Just to name a few:
```

Comment:
(Comment not request) This is another example that calls into question the purpose of these guides -- are they a replacement for the hands-on workshops AND decks OR just the workshops? If just OR workshops, then passages like these a redundant, unnecessary and better handled by the slides and their speaker notes. At most there should be a link to the docs or to refer to the presentation deck.

Reply:
Mmm, understand your statement.  These labs are hit *after* the slides for the corresponding section have been delivered.  However, reinforcing concepts is usually helpful for students to understand and digest the material.  This would be up to @tima to determine if it'll be useful or we should take it out.